### DISCUSSION TITLE
registerFileHandle and spatial extension?

### DISCUSSION THREAD

**Discussion by nshiab** [duckdb/duckdb-wasm#1791] [Q&A]
Hello!

You have great examples showing how to load data from a File handle here: https://duckdb.org/docs/api/wasm/data_ingestion

But this doesn't seem to work with the spatial extension. Here's what I tried to do:

```ts
// file is picked by the user with the File API
await db.registerFileHandle(
  file.name,
  file,
  DuckDBDataProtocol.BROWSER_FILEREADER,
  true
);

await c.query(
  `INSTALL spatial; LOAD spatial;
  CREATE OR REPLACE TABLE table1 AS SELECT * FROM ST_Read(`${file.name}`)`
);
```

But I get this error:
```
IO Error: Unknown file type
```

Any idea what I am doing wrong?

Thank you! 🙏

**Comment by yfyf** [comment]
I have a similar issue, it seems like `ST_Read` is unable to handle any files registered via `registerFile{Text,Buffer,Handle}`.

Here's an example with `registerFileText`:

```js
const geoData = {
  "type": "Feature",
  "geometry": {
    "type": "Point",
    "coordinates": [125.6, 10.1]
  },
  "properties": {
    "name": "Dinagat Islands"
  }
}

const fileName = 'geodata.json'; 
await db.registerFileText(
    fileName,
    JSON.stringify(geoData)
);

const c = await db.connect();
await c.query("INSTALL spatial; LOAD spatial;");
const rez = await c.query(
    `SELECT * FROM ST_Read('${fileName}');`
);
```

which outputs `Error: IO Error: Unknown file type`

`ST_Read` via an HTTP(S) URL works fine. I have also tried to install latest `spatial` from nightly extensions.

Running duckdb-wasm@1.28.1-dev99.0

**Comment by nshiab** [comment]
Since I am not alone with this problem, I created an issue: https://github.com/duckdb/duckdb-wasm/issues/1793
